sqlccl: restoring table desc and namespace entries: unexpected value

While testing #16317, came upon this error while attempting a restore:

```
root@:26257/a> RESTORE d.* FROM 'nodelocal:///home/mjibson/backup' WITH into_db = 'a';
pq: restoring 2 TableDescriptors: restoring table desc and namespace entries: unexpected value: raw_bytes:"O\024\264w\001\206\001" timestamp:<wall_time:1507110108427042933 > 
```

```
$ ./cockroach load show ~/backup
StartTime: 1970-01-01T00:00:00Z (0.000000000,0)
EndTime: 2017-10-04T09:40:44.686049013Z (1507110044.686049013,0)
DataSize: 7586 (7.4 KiB)
Rows: 4
IndexEntries: 0
SystemRecords: 27
FormatVersion: 0
ClusterID: b077a425-c96d-4613-be74-516429108b8c
NodeID: 1
BuildInfo: CockroachDB CCL v1.2-alpha.20170901-547-gb36405997 (linux amd64, built 2017/10/03 04:07:43, go1.9)
Spans:
        /Table/3/{1-2}
        /Table/4/{1-2}
        /Table/60/{1-2}
        /Table/64/{1-2}
Files:
        285211507808665601.sst:
                Span: /Table/3/{1-2}
                Sha512: 4909e904b2cfe230951891fed4599f68f50648e1a420299f73c1f7c97b3c68273ffdc6897a68a204acbcb98b5937b138c6814a253618a3fe420e0651258dd897
                DataSize: 7530 (7.4 KiB)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 27
        285211507807518721.sst:
                Span: /Table/60/{1/2-2}
                Sha512: dba68f338569d9f44a12e013e107232d8e65cb9864c5271080322bcddf19e179cffcafd0ca4f0ce1625b22324573eaf2a2d2af8b42f5eaa9735926ff634353a2
                DataSize: 38 (38 B)
                Rows: 2
                IndexEntries: 0
                SystemRecords: 0
        285211507806928897.sst:
                Span: /Table/64/1{-/2}
                Sha512: 097eae9327f36c786572f2fa01d3c80949ecdaf770b47fab309f9840809efe4f67adbaf71bf80e7a0c3d90da45e63f13036e641f04876f2311287c722c59ab8f
                DataSize: 9 (9 B)
                Rows: 1
                IndexEntries: 0
                SystemRecords: 0
        285211507807944705.sst:
                Span: /Table/64/{1/2-2}
                Sha512: d210610f54424a5c30397fab03621800d37d87c1cf4d6a646b6f8b8de7f444b2dfc6f99ad74868e156da8caee804bfcf335028b53c036d920239dba86d185233
                DataSize: 9 (9 B)
                Rows: 1
                IndexEntries: 0
                SystemRecords: 0
Descriptors:
        1: system (database)
        3: descriptor (table)
        4: users (table)
        59: d (database)
        60: t (table)
        64: t (table)
```

Note there descriptors 60 and 64, both named t. I had previously dropped a table `t` and recreated it, so it appears that when backup is fetching descriptors it also fetches dropped table descriptors.

Here's another failure of this type:

```
func TestBackupRestoreWhileDropping(t *testing.T) {
	defer leaktest.AfterTest(t)()

	const numAccounts = 1
	_, dir, _, sqlDB, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, initNone)
	defer cleanupFn()

	sqlDB.Exec(`DROP DATABASE data CASCADE`)
	sqlDB.Exec(`
		CREATE DATABASE data;
		CREATE TABLE data.t (i int);
		INSERT INTO data.t VALUES (1);
	`)

	if err := verifyBackupRestoreStatementResult(
		sqlDB, "BACKUP DATABASE data TO $1", dir,
	); err != nil {
		t.Fatal(err)
	}

	sqlDB.Exec("CREATE DATABASE data2")

	if err := verifyBackupRestoreStatementResult(
		sqlDB, "RESTORE data.* FROM $1 WITH OPTIONS ('into_db'='data2')", dir,
	); err != nil {
		t.Fatal(err)
	}
}
```

Fails with:

```
--- FAIL: TestBackupRestoreWhileDropping (0.11s)
	backup_test.go:2027: error executing 'BACKUP DATABASE data TO $1': pq: unknown ParentID: 50
```

--- a/pkg/ccl/sqlccl/backup_test.go
+++ b/pkg/ccl/sqlccl/backup_test.go
@@ -2008,3 +2008,46 @@ func TestPointInTimeRecovery(t *testing.T) {
 		sqlDB.CheckQueryResults(`SELECT * FROM data.bank ORDER BY id`, beforeBadThingData)
 	})
 }
+
+func TestBackupRestoreDropDB(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const numAccounts = 1
+	_, dir, _, sqlDB, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, initNone)
+	defer cleanupFn()
+
+	sqlDB.Exec(`DROP DATABASE data CASCADE`)
+	sqlDB.Exec(`
+		CREATE DATABASE data;
+		CREATE TABLE data.bank (i int);
+		INSERT INTO data.bank VALUES (1);
+	`)
+
+	sqlDB.Exec("BACKUP DATABASE data TO $1", dir)
+	sqlDB.Exec("CREATE DATABASE data2")
+	sqlDB.Exec("RESTORE data.* FROM $1 WITH OPTIONS ('into_db'='data2')", dir)
+
+	expected := sqlDB.QueryStr(`SELECT * FROM data.bank`)
+	sqlDB.CheckQueryResults(`SELECT * FROM data2.bank`, expected)
+}
+
+func TestBackupRestoreDropTable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const numAccounts = 1
+	_, dir, _, sqlDB, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, initNone)
+	defer cleanupFn()
+
+	sqlDB.Exec(`DROP TABLE data.bank`)
+	sqlDB.Exec(`
+		CREATE TABLE data.bank (i int);
+		INSERT INTO data.bank VALUES (1);
+	`)
+
+	sqlDB.Exec("BACKUP DATABASE data TO $1", dir)
+	sqlDB.Exec("CREATE DATABASE data2")
+	sqlDB.Exec("RESTORE data.* FROM $1 WITH OPTIONS ('into_db'='data2')", dir)
+
+	expected := sqlDB.QueryStr(`SELECT * FROM data.bank`)
+	sqlDB.CheckQueryResults(`SELECT * FROM data2.bank`, expected)
+}

--- a/pkg/ccl/sqlccl/targets.go
+++ b/pkg/ccl/sqlccl/targets.go
@@ -90,6 +90,9 @@ func descriptorsMatchingTargets(
 
 	for _, desc := range descriptors {
 		if tableDesc := desc.GetTable(); tableDesc != nil {
+			if tableDesc.Dropped() {
+				continue
+			}
 			dbDesc, ok := databasesByID[tableDesc.ParentID]
 			if !ok {
 				return nil, errors.Errorf("unknown ParentID: %d", tableDesc.ParentID)

